primer/test/Tests/Action/Available.hs

```diff
@@ -463,6 +463,12 @@ tasty_available_actions_accepted = withTests 500 $
         (StudentProvided, (Left (ActionError (CaseBranchAlreadyExists (PatPrim _))), _)) -> do
           label "add duplicate primitive case branch"
           annotate "ignoring CaseBranchAlreadyExistsPrim error as was generated constructor"
+        (StudentProvided, (Left (TypeDefAlreadyExists _), _)) -> do
+          pure ()
+        (StudentProvided, (Left (ConAlreadyExists _), _)) -> do
+          pure ()
+        (StudentProvided, (Left (ParamAlreadyExists _), _)) -> do
+          pure ()
```

Comment:
Improving the generators to exercise these clashes would be a good idea.

My current (somewhat scattered) thoughts:
- I'm not sure we will hit `TypeDefAlreadyExists`, since we don't "offer" to create new (type)defs, those global actions are somewhat separate
- `ConAlreadyExists`: when doing this action we `genName`, which often picks from a small set to increase clashes; however, when we initially generate the program, we generate constructor names via `freshNameForCxt`, which has an entirely different mechanism, thus avoiding clashes.
- `ParamAlreadyExists`: this is similar to the case for constructors.